src/utils/fshingTrip.ts

```diff
@@ -0,0 +1,82 @@
+import path from 'path';
+import chalk from 'chalk';
+import { createTwoFilesPatch } from 'diff';
+import { execSync } from 'child_process';
+import temp from 'temp';
+import { union } from 'lodash';
+import fs from 'fs-extra';
+
+import { getFilesRecursive, logger } from '.';
+
+export function fshingTrip(inDir: string, outDir: string, useLocalSUSHI: boolean): void {
+  // Make a pretty box to let the user know we are going into SUSHI mode
+  // NOTE: If we add a box to the end of GoFSH output, it may make sense to modify this
+  // so we don't have double boxes
+  const clr = chalk.green;
+  [
+    clr('╔═════════════════════════════════════════════════════════════════╗'),
+    clr('║') + '             Generating round trip results via SUSHI             ' + clr('║'),
+    clr('╚═════════════════════════════════════════════════════════════════╝')
+  ].forEach(line => console.log(line));
+
+  // If we run with "npx", then the version GoFSH depends on will be used. Otherwise if the
+  // user sets the "use-local-sushi" flag, use their globally installed SUSHI
+  try {
+    execSync(`${useLocalSUSHI ? '' : 'npx '}sushi ${outDir}`, { stdio: 'inherit' });
+  } catch {
+    logger.warn('SUSHI finished with errors, this may affect the resulting comparison');
+  }
+  const inputFilesMap = getFilesMap(inDir);
+  const outputFilesMap = getFilesMap(outDir);
+  const files = union(Array.from(inputFilesMap.keys()), Array.from(outputFilesMap.keys()));
+  files.sort();
+
+  temp.track();
+  const diffFile = temp.openSync({ suffix: '.diff' });
+
+  files.forEach(file => {
+    const inputFilePath = inputFilesMap.get(file) ?? path.join(inDir, file);
+    const outputFilePath = outputFilesMap.get(file) ?? path.join(outDir, file);
+    const inputFileJSON = fs.existsSync(inputFilePath) ? fs.readJSONSync(inputFilePath) : '';
+    const outputFileJSON = fs.existsSync(outputFilePath) ? fs.readJSONSync(outputFilePath) : '';
+
+    // It is best to remove the snapshot, since otherwise it tends to align with the differential in the
+    // diff, creating a pretty meaningless diff
+    if (inputFileJSON.snapshot) {
+      inputFileJSON.snapshot =
+        'NOTE: Snapshots are not generated by SUSHI, so this element is omitted for sake of comparison';
+    }
+
+    const patch = createTwoFilesPatch(
+      path.relative(process.cwd(), inputFilePath),
+      path.relative(process.cwd(), outputFilePath),
+      JSON.stringify(inputFileJSON, null, 2),
+      JSON.stringify(outputFileJSON, null, 2)
+    );
+    if (!/@@/.test(patch)) {
+      // No difference found
+      return;
+    }
+    fs.appendFileSync(diffFile.fd, patch, { encoding: 'utf8' });
+  });
+  try {
+    execSync(
+      `npx diff2html -i file -s side -F fshing-trip-comparison.html --hwt ${path.join(
+        __dirname,
+        'template.html'
+      )} -- ${diffFile.path}`
+    );
+    logger.info(
+      `Generated comparison file to ${path.join(process.cwd(), 'fshing-trip-comparison.html')}`
```

Comment:
I didn't really expect the output to go to the root of the GoFSH project. I was somehow expecting it to go into the `gofsh` or output directory I specify when running GoFSH. Now that I stop and think about that, I'm not sure that it actually does make sense to be there if the idea is that the output directory is set up to be a FSH Tank. But I think I was looking for it with the rest of my output so I could go back and look at the files that were generated all together. That, and I wanted to delete all the remnants of my first GoFSH run in one clean `rm -rf gofsh` swoop. But that might not be a usual move for most users.

So, if we don't want to put the diff file in the output folder and we want to keep it here, how do you feel about putting this path into the `.gitignore`? We're already ignoring `gofsh` by default and it feels in a similar category of files.

Reply:
I think I agree w/ @jafeltra on this.  I'd like to see the file in the output folder.  If the person wants to compare it against the actual generated files, or edit the files to account for unintended differences, I think it's easier to have it there -- and it shouldn't hurt SUSHI to have it there either.  It's a nice record of the process that people might want to keep around for a bit.